### PR TITLE
Calculating appropriate ranks for displaying data engineering layers

### PR DESCRIPTION
**Pull request by andrii-ivaniuk**
## Description

https://jira.quantumblack.com/browse/KED-1412

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Comment by andrii-ivaniuk** [comment]
Input data:
`example_iris_data:
  type: pandas.CSVDataSet
  filepath: data/01_raw/iris.csv
  layer: raw


data_ratio:
  type: pandas.CSVDataSet
  filepath: data/02_intermediate/iris.csv
  layer: intermediate

example_train_x:
  type: pandas.CSVDataSet
  filepath: data/02_intermediate/train_x.csv
  layer: intermediate


example_train_y:
  type: pandas.CSVDataSet
  filepath: data/02_intermediate/train_x.csv
  layer: intermediate


example_test_x:
  type: pandas.CSVDataSet
  filepath: data/02_intermediate/train_x.csv
  layer: layer4


example_test_y:
  type: pandas.CSVDataSet
  filepath: data/02_intermediate/train_x.csv
  layer: prime


prime:
  type: pandas.CSVDataSet
  filepath: data/02_intermediate/train_x.csv
  layer: prime`


Example ranks:
`$ kedro viz
{'data_ratio': 2,
 'dummy': 7,
 'example_iris_data': 0,
 'example_test_x': 6,
 'example_test_y': 8,
 'example_train_x': 4,
 'example_train_y': 4,
 'prime': 8,
 'split_data': 3}
`

File: package/kedro_viz/server.py [hunk 1]
```diff
@@ -195,46 +195,102 @@ def pretty_name(name):
...
-        nodes.append(
-            {
-                "type": "parameters" if is_param else "data",
-                "id": _hash(namespace),
-                "name": pretty_name(namespace),
-                "full_name": namespace,
-                "tags": sorted(tags),
-            }
-        )
+            if data_set.startswith("param"):
```
Comment (andrii-ivaniuk) [inline package/kedro_viz/server.py]: What layer should be used for params?

File: package/kedro_viz/server.py [hunk 2]
```diff
@@ -195,46 +195,102 @@ def pretty_name(name):
...
+                continue
+            ds = catalog_metadata.get(data_set)
+            if ds:
+                input_layers.add(ds._layer)
```
Comment (andrii-ivaniuk) [inline package/kedro_viz/server.py]: I assume that the dataset has a non-empty `layer`. What if the `layer` is not set?

File: package/kedro_viz/server.py [hunk 3]
```diff
@@ -195,46 +195,102 @@ def pretty_name(name):
...
+    #     nodes.append(
+    #         {
+    #             "type": "parameters" if is_param else "data",
+    #             "id": _hash(namespace),
+    #             "name": pretty_name(namespace),
+    #             "full_name": namespace,
+    #             "tags": sorted(tags),
+    #         }
+    #     )
+    #
```
Comment (richardwestenra) [inline package/kedro_viz/server.py]: What's up with this commented-out code? Is this still a work-in-progress, or is it ready for review?

**Comment by andrii-ivaniuk** [comment]
> Hey, worth noting that it looks like we'll need to implement the ranking calculation in the front-end, as it'll need to be able to update on the fly when nodes are hidden.

We decided to arrange a meeting regarding this.

File: package/kedro_viz/server.py (hunk 3 above)
Comment (andrii-ivaniuk) [reply package/kedro_viz/server.py]: It's a draft. I just implemented the code that calculates ranks.

File: package/kedro_viz/server.py (hunk 1 above)
Comment (DmitriiDeriabinQB) [reply package/kedro_viz/server.py]: If the dataset does not have a layer it should still be ranked - thee logic stays the same. The same applies to params, they are treated as datasets without layer.

File: package/kedro_viz/server.py (hunk 2 above)
Comment (DmitriiDeriabinQB) [reply package/kedro_viz/server.py]: Its position is determined by the node it is attached to as an input or output

**Comment by andrii-ivaniuk** [comment]
It will be implemented on UI side.

**Pull request closed without merging**